### PR TITLE
Let use roottest cmake separately from ROOT building

### PR DESCRIPTION
**Pull request by linev**
LZ4_VERSION variable should have meaningful value.

### CODE REVIEW

File: root/io/filemerger/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,9 @@
+if("${LZ4_VERSION}" STREQUAL "")
+   set(lz4_ver 1.8.3)
```
Comment (pcanal) [inline root/io/filemerger/CMakeLists.txt]: Why 1.8.3 (as opposed to any other random version)?  @oshadura outside of ROOT's CMake do we have access to the version of lz4 used/available in ROOT?

Comment (linev) [reply root/io/filemerger/CMakeLists.txt]: 1.8.3 is lz4 version in ROOT master.
To be fully exact one should parse lz4.h header as ROOT cmake does

**Comment by linev** [comment]
That kind of changes, Philippe?
Even random number is better than no number - as now.

**Comment by pcanal** [comment]
Could you write a little root.exe script that grab and print the version number and use that in the CMake file?  Thanks. [Most of us don't use roottest in this mode and we will forget to upgrade that line next time we update zlib ... in addition a lot of people use 'external' zlib libraries which will have different version number.]

**Comment by linev** [comment]
> Could you write a little root.exe script that grab and print the version number and use that in the CMake file? 

It is not that simple. First of all - I do not know if lz4 was build or not. Second, I have no idea which version (builtin or external) was used. And finally - where to search "lz4.h"? I just need to reimplement complete ROOT cmake logic.

And I do not want to test filemerger at all - just to run several tests like `ctest -R json`
Seems to be, there is difference for LZ4_VERSION < 1.7.5. 
Using higher values mean that any new version should be OK

File: root/io/filemerger/CMakeLists.txt (hunk 1 above)
Comment (oshadura) [reply root/io/filemerger/CMakeLists.txt]: I will take a look, it is bug introduced by me :( sorry for this...

**Comment by oshadura** [comment]
I created two separate PRs resolving this issue, sorry for delay!

**Comment by Axel-Naumann** [comment]
@oshadura @linev @pcanal can and should this be merged into master before the v6.18 branch?

**Comment by linev** [comment]
I have workaround and can use `roottest` with minor patches. 
But of course, it is nice to have `roottest` as before - fully independent from ROOT building.

**Comment by oshadura** [comment]
> I have workaround and can use `roottest` with minor patches.
> But of course, it is nice to have `roottest` as before - fully independent from ROOT building.

I have solution in https://github.com/root-project/roottest/pull/319, can you test please?

**Comment by linev** [comment]
> I have solution in #319, can you test please?

No, it does not work - same problem remains. 
And I am not sure if it can - I see `${ROOT_SOURCE_DIR}/` in the modified code. But if I build ROOT, I may not have sources at all.

**Comment by oshadura** [comment]
Can you try again, because I tried and it works for me in case when you source ROOT from build directory and use standalone roottest.git (it is roottest outside of root directory with a separate build directory).

You need to have next line (it is a part of changes): exec_program(root-config ARGS "--srcdir" OUTPUT_VARIABLE ROOT_SOURCE_DIR)

What is your failure? Can you send me a gist?

**Comment by oshadura** [comment]
> > I have solution in #319, can you test please?
> 
> No, it does not work - same problem remains.
> And I am not sure if it can - I see `${ROOT_SOURCE_DIR}/` in the modified code. But if I build ROOT, I may not have sources at all.

What do you mean you don't have sources at all? how do you build ROOT then?

**Comment by linev** [comment]
Here is output:
```
cmake /home/linev/git/roottest                                                                                                                                                                                            (15.05. 12:28:21) !10083 
-- The C compiler identification is GNU 8.2.1
-- The CXX compiler identification is GNU 8.2.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for bitness: Found 64 bit architecture.
-- Found PythonInterp: /usr/bin/python3.6 (found version "3.6.5") 
-- Found PythonLibs: /usr/lib64/libpython2.7.so (found version "2.7.15") 
-- Scanning subdirectories for tests...
CMake Error at root/io/filemerger/CMakeLists.txt:73 (if):
  if given arguments:

    "VERSION_LESS" "1.7.5"

  Unknown arguments specified
```

**Comment by oshadura** [comment]
`oksana@oksana-XPS-13-9380:~$ cd ~/CERN_sources/
oksana@oksana-XPS-13-9380:~/CERN_sources$ source root/builds/bin/thisroot.sh 
oksana@oksana-XPS-13-9380:~/CERN_sources$ cd roottest/builds/
oksana@oksana-XPS-13-9380:~/CERN_sources/roottest/builds$ 
oksana@oksana-XPS-13-9380:~/CERN_sources/roottest/builds$ cmake ..
-- Check for bitness: Found 64 bit architecture.
-- Scanning subdirectories for tests...
-- Configuring done
-- Generating done
-- Build files have been written to: /home/oksana/CERN_sources/roottest/builds
oksana@oksana-XPS-13-9380:~/CERN_sources/roottest/builds$ rm -rf CMakeCache.txt 
oksana@oksana-XPS-13-9380:~/CERN_sources/roottest/builds$ cmake ..
-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for bitness: Found 64 bit architecture.
-- Found PythonInterp: /usr/bin/python (found version "2.7.16") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython2.7.so (found version "2.7.16") 
-- Scanning subdirectories for tests...
-- Configuring done
-- Generating done
-- Build files have been written to: /home/oksana/CERN_sources/roottest/builds
oksana@oksana-XPS-13-9380:~/CERN_sources/roottest/builds$ 
`

**Comment by oshadura** [comment]
Here is my output: https://gist.github.com/oshadura/e630d9f1d2178508394c8f73d3090aa0

**Comment by linev** [comment]
> What do you mean you don't have sources at all? how do you build ROOT then?

In my case sources are there, but in general one could have situation when ROOT was build and install on the system and there are no sources available for it.

**Comment by oshadura** [comment]
> > What do you mean you don't have sources at all? how do you build ROOT then?
> 
> In my case sources are there, but in general one could have situation when ROOT was build and install on the system and there are no sources available for it.

In this case, I merged before fix for such situation and here is a gist:
https://gist.github.com/oshadura/b6aca3cc06a16344e8e36afb7e5032dd

**Comment by oshadura** [comment]
@linev what is a CMake version are you using?

**Comment by linev** [comment]
> what is a CMake version are you using?

```
cmake --version                                                                                                                                                                                                        (15.05. 12:42:49) !10093 1 ↵
cmake version 3.13.1
```

**Comment by oshadura** [comment]
@linev strange, me too! Do you have any  output here `root-config --config | grep LZ4_VERSION`?

**Comment by linev** [comment]
> Do you have any output here root-config --config | grep LZ4_VERSION?

No, empty string. But now I see a problem.
I have lz4 installed on my machine. And root-config **DOES NOT STORE** its version.
Here full output:
```
CFITSIO_INCLUDE_DIR=/usr/include/cfitsio CFITSIO_LIBRARY=/usr/lib64/libcfitsio.so DAVIX_INCLUDE_DIR=/home/linev/build/webgui/builtins/davix/DAVIX-prefix/include/davix DAVIX_INCLUDE_DIRS=/home/linev/build/webgui/builtins/davix/DAVIX-prefix/include/davix DAVIX_LIBRARY=/home/linev/build/webgui/builtins/davix/DAVIX-prefix/lib/libdavix.a FFTW_INCLUDE_DIR=/usr/include FFTW_LIBRARY=/usr/lib64/libfftw3.so FREETYPE_INCLUDE_DIR_freetype2=/usr/include/freetype2 FREETYPE_INCLUDE_DIR_ft2build=/usr/include/freetype2 FREETYPE_LIBRARY_RELEASE=/usr/lib64/libfreetype.so GIF_INCLUDE_DIR=/usr/include GIF_LIBRARY=/usr/lib64/libgif.so GSL_CBLAS_LIBRARY=/usr/lib64/libgslcblas.so GSL_CBLAS_LIBRARY_DEBUG=/usr/lib64/libgslcblas.so GSL_INCLUDEDIR=/usr/include GSL_INCLUDE_DIR=/usr/include GSL_INCLUDE_DIRS=/usr/include GSL_LIBRARY=/usr/lib64/libgsl.so GSL_LIBRARY_DEBUG=/usr/lib64/libgsl.so GSL_VERSION=2.5 JPEG_INCLUDE_DIR=/usr/include JPEG_LIBRARY=/usr/lib64/libjpeg.so JPEG_LIBRARY_RELEASE=/usr/lib64/libjpeg.so LIBXML2_INCLUDE_DIR=/usr/include/libxml2 LIBXML2_LIBRARY=/usr/lib64/libxml2.so LZ4_INCLUDE_DIR=/usr/include LZ4_LIBRARY=/usr/lib64/liblz4.so LZMA_INCLUDE_DIR=/usr/include LZMA_LIBRARY=/usr/lib64/liblzma.so MYSQL_INCLUDE_DIR=/usr/include/mysql OPENGL_EGL_INCLUDE_DIR=/usr/include OPENGL_GLX_INCLUDE_DIR=/usr/include OPENGL_INCLUDE_DIR=/usr/include OPENGL_egl_LIBRARY=/usr/lib64/libEGL.so OPENGL_glu_LIBRARY=/usr/lib64/libGLU.so OPENGL_glx_LIBRARY=/usr/lib64/libGLX.so OPENGL_opengl_LIBRARY=/usr/lib64/libOpenGL.so OPENSSL_CRYPTO_LIBRARY=/usr/lib64/libcrypto.so OPENSSL_INCLUDE_DIR=/usr/include OPENSSL_SSL_LIBRARY=/usr/lib64/libssl.so PCRE_INCLUDE_DIR=/usr/include PCRE_PCRE_LIBRARY_DEBUG=/usr/lib64/libpcre.so PCRE_PCRE_LIBRARY_RELEASE=/usr/lib64/libpcre.so PC_LIBXML_INCLUDEDIR=/usr/include PC_LIBXML_INCLUDE_DIRS=/usr/include/libxml2 PC_LIBXML_LIBRARIES=xml2 PC_LIBXML_STATIC_INCLUDE_DIRS=/usr/include/libxml2 PC_LIBXML_VERSION=2.9.8 PC_SQLITE_INCLUDEDIR=/usr/include PC_SQLITE_LIBRARIES=sqlite3 PC_SQLITE_VERSION=3.25.2 PNG_LIBRARY=/usr/lib64/libpng.so PNG_LIBRARY_RELEASE=/usr/lib64/libpng.so PNG_PNG_INCLUDE_DIR=/usr/include PYTHON_INCLUDE_DIR=/usr/include/python2.7 PYTHON_LIBRARY=/usr/lib64/libpython2.7.so PYTHON_LIBRARY_RELEASE=/usr/lib64/libpython2.7.so SQLITE_INCLUDE_DIR=/usr/include SQLITE_LIBRARIES=/usr/lib64/libsqlite3.so TIFF_INCLUDE_DIR=/usr/include TIFF_LIBRARY=/usr/lib64/libtiff.so TIFF_LIBRARY_RELEASE=/usr/lib64/libtiff.so UUID_INCLUDE_DIR=/usr/include UUID_LIBRARY=uuid VDT_INCLUDE_DIRS=/home/linev/build/webgui/include VDT_LIBRARIES=/home/linev/build/webgui/lib/libvdt.so X11_ICE_INCLUDE_PATH=/usr/include X11_SM_INCLUDE_PATH=/usr/include X11_X11_INCLUDE_PATH=/usr/include X11_XRes_INCLUDE_PATH=/usr/include X11_XShm_INCLUDE_PATH=/usr/include X11_XSync_INCLUDE_PATH=/usr/include X11_XTest_INCLUDE_PATH=/usr/include X11_Xaccess_INCLUDE_PATH=/usr/include X11_Xaccessrules_INCLUDE_PATH=/usr/include X11_Xaccessstr_INCLUDE_PATH=/usr/include X11_Xau_INCLUDE_PATH=/usr/include X11_Xcomposite_INCLUDE_PATH=/usr/include X11_Xcursor_INCLUDE_PATH=/usr/include X11_Xdamage_INCLUDE_PATH=/usr/include X11_Xdmcp_INCLUDE_PATH=/usr/include X11_Xfixes_INCLUDE_PATH=/usr/include X11_Xft_INCLUDE_PATH=/usr/include X11_Xi_INCLUDE_PATH=/usr/include X11_Xinerama_INCLUDE_PATH=/usr/include X11_Xinput_INCLUDE_PATH=/usr/include X11_Xkb_INCLUDE_PATH=/usr/include X11_Xkbfile_INCLUDE_PATH=/usr/include X11_Xkblib_INCLUDE_PATH=/usr/include X11_Xlib_INCLUDE_PATH=/usr/include X11_Xmu_INCLUDE_PATH=/usr/include X11_Xpm_INCLUDE_PATH=/usr/include X11_Xrandr_INCLUDE_PATH=/usr/include X11_Xrender_INCLUDE_PATH=/usr/include X11_Xscreensaver_INCLUDE_PATH=/usr/include X11_Xshape_INCLUDE_PATH=/usr/include X11_Xt_INCLUDE_PATH=/usr/include X11_Xutil_INCLUDE_PATH=/usr/include X11_Xv_INCLUDE_PATH=/usr/include X11_dpms_INCLUDE_PATH=/usr/include X11_xf86misc_INCLUDE_PATH=/usr/include X11_xf86vmode_INCLUDE_PATH=/usr/include ZLIB_INCLUDE_DIR=/usr/include ZLIB_LIBRARY_RELEASE=/usr/lib64/libz.so xxHash_INCLUDE_DIR=/home/linev/git/webgui/builtins/xxhash xxHash_INCLUDE_DIRS=/home/linev/git/webgui/builtins/xxhash xxHash_LIBRARIES=xxHash::xxHash xxHash_LIBRARY=$<TARGET_FILE:xxhash> xxHash_VERSION=0.6.4 xxHash_VERSION_STRING=0.6.4
```

Only when ROOT used build-in lz4, it also provides LZ4_VERSION - I can see it on other machine.

**Comment by oshadura** [comment]
Ok I will test this case and fix it, it should be very simple fix.

**Comment by amadio** [comment]
The fix for this cannot involve changes in `root.git`, that's the wrong approach. Please either call `find_package(LZ4)` in roottest (non-builtin) case, or use `root.exe` to find out, like this:
```sh
$ root.exe -l -q -e "#include <lz4.h>" -e "cout << LZ4_versionString();"

1.8.3
```

**Comment by amadio** [comment]
Also, we should always assume that the ROOT sources are gone, as if we want to test an installed release in CVMFS (i.e. an LCG build), a distribution package installed with conda, yum, etc, the source directory will be long gone.

**Comment by linev** [comment]
@amadio 

> $ root.exe -l -q -e "#include <lz4.h>" -e "cout << LZ4_versionString();"

You are using `#include <lz4.h>` without any extra cxx flags. 
Which include path will be used?
If I used some special `LZ4_INCLUDE_DIR` in ROOT cmake, will it be used here as well?

**Comment by oshadura** [comment]
@amadio, what do you mean? I don't understand your comment. What we are trying to do is to fix possibility to use roottest as standalone project. I can't call find_package(LZ4), because we need then add FindLZ4.cmake in roottest. But still there is a big probability that we will not know with LZ4 was used in the ROOT actually (since lz4 versions are different).

**Comment by linev** [comment]
> The fix for this cannot involve changes in root.git

Why? Why not to add LZ4_VERSION into `root-config --config` output.
If it not there - put some predefined value which is better than nothing

**Comment by oshadura** [comment]
@amadio recommended me alternative fix in case of system LZ4. I will update PR accordingly.

**Comment by amadio** [comment]
@linev If you bake `LZ4_VERSION` into `root-config`, then when you update LZ4 in your system the test will start failing because it will check for the wrong references.

**Comment by linev** [comment]
> If you bake LZ4_VERSION into root-config, then when you update LZ4 in your system the test will start failing because it will check for the wrong references.

But in this case I also have to recompile ROOT - there is not way around.
Or version change is minimal and not relevant for roottest.

**Comment by amadio** [comment]
If, on the other hand, lz4 is builtin, then the version is baked into ROOT, and we have to query the version through ROOT, maybe not in the way I did, but by somehow calling `LZ4_versionString()` that will resolve from the builtin lz4 library.

**Comment by amadio** [comment]
> But in this case I also have to recompile ROOT - there is not way around

That's not true, you can safely update LZ4 and not recompile ROOT and it will work normally. You only have to recompile when a library introduces incompatible changes to ABI/API, and for C libraries that's rare. You can update X11, OpenGL, libafterimage, MySQL, etc, even glibc, and you will almost never need to recompile ROOT.

**Comment by amadio** [comment]
`root-config --config` is just saving CMake variables that will inevitably point to non-existent directories/files as soon as you install ROOT and both source and build directories no longer exist. We need to use other alternatives.

**Comment by linev** [comment]
> root-config --config is just saving CMake variables that will inevitably point to non-existent directories/files as soon as you install ROOT and both source and build directories no longer exist. We need to use other alternatives.

But exactly this the reason why `root-config --config` is only solution for us.
Example - you build and install ROOT with built-in lz4. 
All sources are gone, lz4.h is gone, how I should ask for lz4 version which was used when building ROOT? If  you change lz4 behind the scene - it is your fault, not the ROOT.

**Comment by oshadura** [comment]
> @amadio recommended me alternative fix in case of system LZ4. I will update PR accordingly.

I updated only way how we find LZ4 for roottest if ROOT is build with system lz4. 
https://github.com/root-project/roottest/pull/319

**Comment by linev** [comment]
@oshadura 
I am not sure that your solution is better. You are trying to parse lz4.h, using 

```
if(LZ4_INCLUDE_DIR AND EXISTS "${LZ4_INCLUDE_DIR}/lz4.h")
```

Does `LZ4_INCLUDE_DIR` parameters comes from root-config? 
If yes, solution is error prone. 
`root-config --config` also includes LZ4_INCLUDE_DIR for built-in version.
You checking directory, where my ROOT sources were when I build ROOT. 
If I did it 2 years ago and 1000 tmes update ROOT source but did not update installation? 

We need to record lz4 version when ROOT was build.

**Comment by oshadura** [comment]
1. No LZ4_INCLUDE_DIR  is not coming from root-config. I am checking **ONLY** LZ4_VERSION.
2. I tested it in the environment and it works, it finds system LZ4 (which has different version) comparing to builtin version.

**Comment by linev** [comment]
> No LZ4_INCLUDE_DIR is not coming from root-config. I am checking ONLY LZ4_VERSION.

But this is just another problem. If I have 10 different LZ4 libraries installed on my system and use very special version with the ROOT and ROOT **still linked** against this special version, your macro fail. With large effort you get some version, which has nothing to do with reality.

Again - why not to write LZ4_VERSION into root-config?
It is already there for built-in case, do it also for other case.

**Comment by oshadura** [comment]
I don't have preferences, for me both approach could work, but both of them have configuration issues in case of very "exotic" setups. I will say if there will be a need I know how to adjust a problem. 
I hope current fix solved your issue!

**Comment by amadio** [comment]
Here is a preliminary build of roottest using LCG nightlies of ROOT: https://epsft-jenkins.cern.ch/view/ROOT/job/roottest-lcg-nightly. It currently fails because `$ROOTSYS` isn't set (it shouldn't need to be set if root.exe is in the `$PATH` or `find_package(ROOT)` can be called).

**Comment by oshadura** [comment]
I will adjust it, thanks!

**Comment by linev** [comment]
#319 solved problem for now

**Comment by amadio** [comment]
I will be happy when this `Dockerfile` can build and run the tests:
```Dockerfile
FROM fedora

RUN dnf update  -y                           \
 && dnf install -y --allowerasing            \
    cmake git root root-* python{2,3}-root   \
 && dnf autoremove -y

RUN git clone http://root.cern/git/roottest  \
 && ctest --build-generator 'Unix Makefiles' \
          --build-and-test   roottest build  \
          --test-command ctest
```
Here's what I get:
```bash
$ docker build .
Sending build context to Docker daemon  3.072kB
Step 1/3 : FROM fedora
 ---> d09302f77cfc
Step 2/3 : RUN dnf update  -y                            && dnf install -y --allowerasing                cmake git root root-* python{2,3}-root    && dnf autoremove -y
 ---> Using cache
 ---> 55ab52ef2bcb
Step 3/3 : RUN git clone http://root.cern/git/roottest   && ctest --build-and-test   roottest build            --build-generator 'Unix Makefiles' 	  --test-command ctest
 ---> Running in 016cfc3ecd36
Cloning into 'roottest'...
Internal cmake changing into directory: /build
Error: cmake execution failed
The C compiler identification is GNU 8.3.1
The CXX compiler identification is GNU 8.3.1
Check for working C compiler: /usr/bin/cc
Configuring
Check for working C compiler: /usr/bin/cc -- works
Detecting C compiler ABI info
Configuring
Detecting C compiler ABI info - done
Detecting C compile features
Configuring
Configuring
Configuring
Detecting C compile features - done
Check for working CXX compiler: /usr/bin/c++
Configuring
Check for working CXX compiler: /usr/bin/c++ -- works
Detecting CXX compiler ABI info
Configuring
Detecting CXX compiler ABI info - done
Detecting CXX compile features
Configuring
Detecting CXX compile features - done
CMake Error at CMakeLists.txt:38 (message):
  ROOTSYS is not set!

Configuring incomplete, errors occurred!
See also "/build/CMakeFiles/CMakeOutput.log".

The command '/bin/sh -c git clone http://root.cern/git/roottest   && ctest --build-and-test   roottest build            --build-generator 'Unix Makefiles' 	  --test-command ctest' returned a non-zero code: 1
```

**Comment by oshadura** [comment]
> I will be happy when this `Dockerfile` can build and run the tests:
> 
> ```dockerfile
> FROM fedora
> 
> RUN dnf update  -y                           \
>  && dnf install -y --allowerasing            \
>     cmake git root root-* python{2,3}-root   \
>  && dnf autoremove -y
> 
> RUN git clone http://root.cern/git/roottest  \
>  && ctest --build-generator 'Unix Makefiles' \
>           --build-and-test   roottest build  \
>           --test-command ctest
> ```
> 
> Here's what I get:
> 
> ```shell
> $ docker build .
> Sending build context to Docker daemon  3.072kB
> Step 1/3 : FROM fedora
>  ---> d09302f77cfc
> Step 2/3 : RUN dnf update  -y                            && dnf install -y --allowerasing                cmake git root root-* python{2,3}-root    && dnf autoremove -y
>  ---> Using cache
>  ---> 55ab52ef2bcb
> Step 3/3 : RUN git clone http://root.cern/git/roottest   && ctest --build-and-test   roottest build            --build-generator 'Unix Makefiles' 	  --test-command ctest
>  ---> Running in 016cfc3ecd36
> Cloning into 'roottest'...
> Internal cmake changing into directory: /build
> Error: cmake execution failed
> The C compiler identification is GNU 8.3.1
> The CXX compiler identification is GNU 8.3.1
> Check for working C compiler: /usr/bin/cc
> Configuring
> Check for working C compiler: /usr/bin/cc -- works
> Detecting C compiler ABI info
> Configuring
> Detecting C compiler ABI info - done
> Detecting C compile features
> Configuring
> Configuring
> Configuring
> Detecting C compile features - done
> Check for working CXX compiler: /usr/bin/c++
> Configuring
> Check for working CXX compiler: /usr/bin/c++ -- works
> Detecting CXX compiler ABI info
> Configuring
> Detecting CXX compiler ABI info - done
> Detecting CXX compile features
> Configuring
> Detecting CXX compile features - done
> CMake Error at CMakeLists.txt:38 (message):
>   ROOTSYS is not set!
> 
> Configuring incomplete, errors occurred!
> See also "/build/CMakeFiles/CMakeOutput.log".
> 
> The command '/bin/sh -c git clone http://root.cern/git/roottest   && ctest --build-and-test   roottest build            --build-generator 'Unix Makefiles' 	  --test-command ctest' returned a non-zero code: 1
> ```

Ok, I am taking a look now.

**Comment by oshadura** [comment]
@amadio $ROOTSYS was always required as a part of roottest setup, it is not my changes. What do you expect to have here instead?

**Comment by amadio** [comment]
The ROOTSYS part is easily fixed with a line like
```
ENV ROOTSYS=/usr
```
The following errors are harder to fix (the one about not finding RootNewMacros, for example).

I didn't say any problems were your or Sergey's fault, but it still doesn't really work even after the changes.

**Comment by linev** [comment]
> ENV ROOTSYS=/usr

I guess, is much better to use:

    ENV ROOTSYS=`root-config --prefix`

**Comment by amadio** [comment]
If `root-config` is already in the PATH, why would you need to export `ROOTSYS`?

**Comment by linev** [comment]
> If root-config is already in the PATH, why would you need to export ROOTSYS?

No idea.
 In my mind, one `root-config` should always be in the PATH before starting roottest. 
Anything else makes no sense.

**Pull request closed without merging**